linux/LinuxMachine.c

```diff
@@ -497,11 +497,13 @@ static void LinuxMachine_scanCPUTime(LinuxMachine* this) {
 
    this->period = (double)this->cpuData[0].totalPeriod / super->activeCPUs;
 
-   char buffer[PROC_LINE_LENGTH + 1];
-   while (fgets(buffer, sizeof(buffer), file)) {
-      if (String_startsWith(buffer, "procs_running")) {
-         this->runningTasks = strtoul(buffer + strlen("procs_running"), NULL, 10);
-         break;
+   if (!ferror(file) && !feof(file)) {
+      char buffer[PROC_LINE_LENGTH + 1];
+      while (fgets(buffer, sizeof(buffer), file)) {
```

Comment:
Wouldn't this still need the `ferror` and `feof` checks?

Reply:
No, because we would only repeatedly call `fgets` in case the previous `fgets` call returned non-`NULL`, i.e. success.